File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    for (let h of [...Array(Math.floor(Math.sqrt(yellow)) + 1)].map((v,i) => i)){
        if (yellow % h !== 0) continue;
        let w = yellow / h;
        
        if (2* (w + h) + 4 === brown) return [w + 2, h + 2]
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
import math
import sys
sys.setrecursionlimit(10 ** 9)

def recursive(k, yellow, brown):
    if yellow % k == 0 and 2 * ((yellow // k) + k) + 4 == brown:
        return k
    
    return recursive(k + 1, yellow, brown)
    
def solution(brown, yellow):
    k = recursive(math.ceil(math.sqrt(yellow)), yellow, brown)
    return [k + 2, yellow // k + 2]